core/src/main/java/org/geysermc/geyser/translator/protocol/java/JavaCustomPayloadTranslator.java

```diff
@@ -72,12 +72,18 @@ public void translate(GeyserSession session, ClientboundCustomPayloadPacket pack
             String dataString = new String(data, 3, data.length - 3, Charsets.UTF_8);
 
             Form form = Forms.fromJson(dataString, type, (ignored, response) -> {
-                byte[] raw = response.getBytes(StandardCharsets.UTF_8);
-                byte[] finalData = new byte[raw.length + 2];
-
-                finalData[0] = data[1];
-                finalData[1] = data[2];
-                System.arraycopy(raw, 0, finalData, 2, raw.length);
+                byte[] finalData;
+                if (response != null) {
+                    byte[] raw = response.getBytes(StandardCharsets.UTF_8);
+                    finalData = new byte[raw.length + 2];
+
+                    finalData[0] = data[1];
+                    finalData[1] = data[2];
+                    System.arraycopy(raw, 0, finalData, 2, raw.length);
+                } else {
+                    // Only need to send the form id
```

Comment:
I would add a note here saying that the form data can be nullable as of 1.19.20.